doc/api.md

```diff
@@ -955,6 +966,121 @@ The IOTA processes the entity attributes looking for a `TimeInstant` attribute.
+
+Once the command request is issued to the IoT agent, the command is stored waiting to be retrieved by the device. In that moment, the status of the command is `"<command>_status": "PENDING"`. 
+
+For HTTP devices, in order to retrieve a poll command from IoTA-JSON Agent the device should make the following request:
+
+```
+GET /iot/json?k=<apikey>&i=<deviceId>&getCmd=1
+Accept: application/json
+```
+
+**FIXME**: example of response to this request should be included, based on the ping example shown above.
+
+For IoT Agents different from IoTA-JSON it is exactly the same just changing in the request the resource, `/iot/json` by the corresponding resource employed by the agent (i.e., IoTA-UL uses `/iot/d` as default resource) and setting the correct `<apikey>` and `<deviceId>`.
+
+It can be also possible for a device to retrieve the commands from the IoT Agent when it sends an observation. It just be needed to include the `&getCmd=1` parameter in the observation request. In the following example a device sends an JSON observation and retrieves the commands from the IoTA-JSON Agent.
+
+```
+POST /iot/json?k=<apikey>&i=<deviceId>&getCmd=1
+Content-Type: application/json
+
+{"t":25,"h":42,"l":"1299"}
+```
+
+This is also possible for IoTA-UL Agent changing in the request the resource, setting the correct `<apikey>`, `<deviceId>`, payload and headers.
+
+Once the command is retrieved by the device the status is updated to `"<command>_status": "DELIVERED"`.
+
+
+#### MQTT devices
+
+For MQTT devices, it is not needed to declare an endpoint (i.e. if included in the provisioning request, it is not used). The device is supposed to be subscribed to the 
+following MQTT topic:
+
+```
+/<apiKey>/<deviceId>/cmd
+```
+
+**FIXME**: example of payload sent to this topic by the IoTA-JSON should be included, based on the ping example shown above.
+
+where it will receive the command information. Please note that the device should subscribe to the broker 
+using the disabled clean session mode (enabled using `--disable-clean-session` option CLI parameter in 
+`mosquitto_sub`). This option means that all of the subscriptions for the device will be maintained after 
+it disconnects, along with subsequent QoS 1 and QoS 2 commands that arrive. When the device reconnects, it
+will receive all of the queued commands.
+
+### Command confirmation
+
+Once the command is completely processed by the device, it should return the result of the command to the IoT 
+Agent. This result will be progressed to the Context Broker where it will be stored in the `<command>_info` 
+attribute. The status of the command will be stored in the `<command>_status` attribute (`OK` if everything 
+goes right). 
+
+#### HTTP
+
+**Polling**
+Eventually, once the device makes the response request with the result of the command the status is updated 
+to`"<command>_status": "OK"`. Also the result of the command delivered by the device is stored in the `<command>_info` attribute.
```

Comment:
It seems the same way apply to both "push" and "pooling" commands, so the "Polling" text in this section is confusing...

Reply:
Maybe related https://github.com/telefonicaid/iotagent-node-lib/issues/1559 ?